src/main/java/br/edu/ufcg/computacao/eureca/backend/api/http/exceptions/pre_enrollment/StudentNotFoundException.java

```diff
@@ -0,0 +1,12 @@
+package br.edu.ufcg.computacao.eureca.backend.api.http.exceptions.pre_enrollment;
+
+import br.edu.ufcg.computacao.eureca.common.exceptions.EurecaException;
+
+public class StudentNotFoundException extends EurecaException {
+
+    private static final String STUDENT_NOT_FOUND = "O estudante com matrícula %s não está cadastrado no sistema.";
```

Comment:
All strings should be in a constant class. This facilitates reviewing error messages without the need to inspect the whole code.

Reply:
Ok, but why not define StudentNotFoundException as an extension of br.edu.ufcg.computacao.eureca.common.exceptions.InvalidParameterException?

This code should be:

package br.edu.ufcg.computacao.eureca.backend.api.http.exceptions;

import br.edu.ufcg.computacao.eureca.common.exceptions.InvalidParameterException;
import br.edu.ufcg.computacao.eureca.common.constants.Messages;

public class StudentNotFoundException extends InvalidParameterException {

   public StudentNotFoundException(String registration) {
        super(String.format(Messages.STUDENT_NOT_FOUND, registration));
   }
}

And:
    private static final String STUDENT_NOT_FOUND = "O estudante com matrícula %s não está cadastrado no sistema.";

Should be defined at br.edu.ufcg.computacao.eureca.common.constants.Messages.

Reply:
Understood, changed